types/utils.go

```diff
@@ -169,15 +23,15 @@ func expandEnvPath(path string, rootlessUID int) (string, error) {
 	return newpath, nil
 }
 
-func DefaultConfigFile(rootless bool) (string, error) {
+func DefaultConfigFile() (string, error) {
 	if defaultConfigFileSet {
 		return defaultConfigFile, nil
 	}
 
 	if path, ok := os.LookupEnv(storageConfEnv); ok {
 		return path, nil
 	}
-	if !rootless {
+	if unshare.GetRootlessUID() == 0 {
```

Comment:
Again on the `IsRootless` vs. `UID == 0` — all Podman callers AFAICS used to call this with `rootless.IsRootless()`, so to ignorant me this looks like a change. Is that intentional?

… and there is also `defaultStoreOptionsIsolated` conditioned on `rootless && rootlessUID != 0`; so we have _three_ different conditions around config files in this subpackage. Are they really intentionally different?

If the logic should not be just a plain `rootless.IsRootless()`, I’d like to see that condition consolidated into a single function (`usePerUserStorage`??? possibly 1 line of code + 10 lines of comments) called everywhere here, so that the individual places don’t drift.

It’s also quite possible I am completely missing the point.

Reply:
no you are right.  I've added a `usePerUserStorage` function that explains why we don't want to use directly `IsRootless`